pilot/pkg/serviceregistry/kube/controller/ambient/multicluster.go

```diff
@@ -336,11 +298,114 @@ func (a *index) buildGlobalCollections(
 		opts,
 	)
 
-	WorkloadAddressIndex := krt.NewIndex[networkAddress, model.WorkloadInfo](GlobalWorkloads, "networkAddress", networkAddressFromWorkload)
-	WorkloadServiceIndex := krt.NewIndex[string, model.WorkloadInfo](GlobalWorkloads, "service", func(o model.WorkloadInfo) []string {
+	workloadNetworkServiceIndex := krt.NewIndex[string, model.WorkloadInfo](GlobalWorkloads, "network;service", func(o model.WorkloadInfo) []string {
+		res := make([]string, 0, len(o.Workload.Services))
+		for svc := range o.Workload.Services {
+			res = append(res, strings.Join([]string{o.Workload.Network, svc}, ";"))
+		}
+		return res
+	})
+
+	coalescedWorkloads := krt.NewManyCollection(
+		workloadNetworkServiceIndex.AsCollection(),
+		func(ctx krt.HandlerContext, i krt.IndexObject[string, model.WorkloadInfo]) []model.WorkloadInfo {
+			parts := strings.Split(i.Key, ";")
```

Comment:
Don't love this; should we just create a small private struct instead? Would remove the splitting logic

Reply:
I don't love it either, but to do `AsCollection`, keys have to be strings (krt/index.go:112). Reading it again, we could use `WithIndexCollectionFromString`, but either way, we need to have splitting/encoding logic.

Reply:
Ah I see; that makes sense. Would be clearer if we dind't need the index I think